config/initializers/apipie.rb

```diff
@@ -173,3 +173,29 @@ def expected_type
     :any_type
   end
 end
+
+class DelayedEnumValidator < Apipie::Validator::BaseValidator
+  def initialize(param_description, block)
+    super(param_description)
+    @block = block
+  end
+
+  def self.build(param_description, argument, options, block)
+    if argument == :delayed_enum
+      new(param_description, options[:of])
```

Comment:
Can we raise an exception if `options[:of]` isn't set?

Also, why is `block` an argument but not used/passed on?

Reply:
This is because of https://github.com/Apipie/apipie-rails/blob/master/lib/apipie/validator.rb#L35. The signature of `self.build` must be the same.